src/test/suite/participant/participant.test.ts

```diff
@@ -129,8 +115,144 @@ suite('Participant Controller Test Suite', function () {
     );
   });
 
+  suite('when not connected', function () {
+    let connectWithConnectionIdStub;
+    let connectWithURIStub;
+
+    beforeEach(function () {
+      connectWithConnectionIdStub = sinon.stub(
+        testParticipantController._connectionController,
+        'connectWithConnectionId'
+      );
+      connectWithURIStub = sinon.stub(
+        testParticipantController._connectionController,
+        'connectWithURI'
+      );
+      sinon.replace(
+        testParticipantController._connectionController,
+        'getActiveDataService',
+        () => null
+      );
+      sinon.replace(
+        testParticipantController._storageController,
+        'get',
+        sinon.fake.returns(true)
+      );
+      sinon
+        .stub(testParticipantController._connectionController, '_connections')
```

Comment:
Not a blocker, I think if possible we should avoid stubbing the interior private aspects of another module, it can increase the friction for changes and lead to some hard to track down testing issue. Instead we can stub what the public/exposed functions return. - Or, even better, avoid stubbing all together (maybe not feasible here).